### ISSUE TITLE
Are CSS break-before/after indicators page break locators?

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag#2259]
There are a few ways in CSS to indicate where to insert / where not to insert a page break when printing. Looking at the definition of page break locators, it seems like they might apply. However the understanding document / technique doesn't mention them. Here's an example of one:

```html
<h1>Hello world</h1>
<p>Lorem ipsum dolor sit amet consectetur?</p>
<h2>More stuff</h2>
<p>Lorem ipsum dolor sit amet consectetur?</p>
<h2>More stuff</h2>
<p>Lorem ipsum dolor sit amet consectetur?</p>
<style>
  h2 { break-before: page }
</style>
```

It's not clear to me why this SC should apply to CSS breaks, however looking at the normative language I think it does. Could this be clarified?

**Comment by patrickhlauke** [comment]
just to backfill/provide context, "this SC" is https://www.w3.org/WAI/WCAG22/Understanding/page-break-navigation.html

I *think* this would indeed count as a page break locator, but as it's arguably variable, I fail to envisage how a site could provide navigation here (other than providing a navigation directly to every `h2`?)

**Comment by WilcoFiers** [comment]
So, apparently this was answered: https://www.w3.org/standards-guidelines/act/rules/73f2c2/ I would ask that the group revisit this issue though. I can't figure out why CSS break-before/after wouldn't apply, and if it's not in the understanding document I don't imagine I'll be the only person to struggle with this.

**Comment by bruce-usab** [comment]
@WilcoFiers please double check the URL we should be looking at.  [73f2c2](https://www.w3.org/WAI/standards-guidelines/act/rules/73f2c2/) is <q>autocomplete attribute has valid value</q>.

I have not used the contemporary CSS syntax docs.  Is this it?
- https://developer.mozilla.org/en-US/docs/Web/CSS/break-before

I did look at the [CSS2](https://www.w3.org/TR/CSS22/page.html#page-breaks) equivalent back in the day.  Back it the day, I thought of it as widow/orphan control.  Re-reading just now, I still think that is true.

IMHO, the CSS `break-before:page` does not meet our definition for [page break locators](https://www.w3.org/WAI/WCAG22/Understanding/page-break-navigation.html#dfn-page-break-locators).

**Comment by alastc** [comment]
I think the relevant bit is the definition of page-break-locators:
> programmatically determinable destination markers that are arranged in a meaningful sequence to represent a locator serving the same purpose as page breaks in a printed document.

Using CSS break-before/after is an instruction going the other way, from the CSS to the print output. It isn't saying that this point is where another version (e.g. printed) has the page break. 

Anything that varies due to the nature of HTML/CSS isn't going to match, it needs to be a point in the HTML (not CSS) with an appropriate role.

I suppose if you had another edition where you carefully made sure the pages always broke before an H2, you could implement it that way, but I don't think that generates a requirement for everyone using a CSS page-break.

**Comment by WilcoFiers** [comment]
> It isn't saying that this point is where another version (e.g. printed) has the page break.

The definition doesn't say that

>  it needs to be a point in the HTML (not CSS) with an appropriate role

Definition doesn't say that either.


I'll gladly accept that CSS break-before/after shouldn't be considered page break locators. But I don't think that's obvious from either the definition or the understanding document. If this is what is meant, our definition should say so, and our understanding document should explain it.

**Comment by alastc** [comment]
> > It isn't saying that this point is where another version (e.g. printed) has the page break.
> 
> The definition doesn't say that

That's how I read "to represent a locator serving the same purpose as page breaks in a printed document."

Particularly with the examples that follow (as part of the definition).

Like Bruce, I'm not sure how to tighten that up without being very technology focused (e.g. a locator with a role of X), which would also narrow it down to one specific technique.

Happy with the suggested note, although we might want to re-factor the 4 notes into standard content?!

**Comment by WilcoFiers** [comment]
> Note: Many print-oriented formatting effects (for example 'break-before:page' in CSS) are not destination markers and do not fit the definition for page break locators

Perhaps the difficulty here is in the phrase "destination marker". I have to admit I don't really understand what that phrase is intended to mean. I don't think I really understand why `role="doc-pagebreak"` would be a destination marker, and how such a role is different from some indicator in CSS, or whatever GDocs uses under the hood to determine where one page ends and the next starts.

**Comment by alastc** [comment]
The difference is you can't link to a CSS page break, it has no concept of location, it isn't a target.

Something with an ID (and appropriate role) makes it a target for navigation, where as a CSS page break is saying to a print mechanism where to paginate, not what number / target it is.

**Comment by WilcoFiers** [comment]
> The difference is you can't link to a CSS page break, it has no concept of location, it isn't a target.

I don't understand how that's different. You can't link to an element with a role attribute any more than you can to an element with a break-after CSS prop. In both cases the element needs an ID if you want to link to it.

**Comment by alastc** [comment]
Based on a [comment on another thread](https://github.com/w3c/wcag/issues/2274#issuecomment-1076827611), would this help:

"programmatically determinable destination markers that represent page breaks in an alternative version of the document."

CCing @mbgower who overhauled the previous version.

These page-breaks may or may not line up with how the HTML doc you are evaluating prints out, but the key is what they represent.

**Comment by alastc** [comment]
In the meeting we resolving to reject the definition but make updates to the understanding doc.

See Wilco's (and Michael's?) comments in https://www.w3.org/2002/09/wbs/35422/wcag22-page-break-nav/results#xq13
